src/TabBarItem.tsx

```diff
@@ -202,7 +209,7 @@ export default class TabBarItem<T extends Route> extends React.Component<
       scrollEnabled === true;
 
     const tabContainerStyle: ViewStyle = {};
-    const itemStyle = isWidthSet ? { width: tabWidth } : null;
+    const itemStyle = isWidthSet && !dynamicWidth ? { width: tabWidth } : null;
```

Comment:
We should use `width: 'auto'` in `tabStyle` for this

Reply:
can null be totally removed in favour of `{width: 'auto'}`?

I've totally removed `dynamicWidth` prop here and it seems to work.

Reply:
```suggestion
const itemStyle = isWidthSet ? { width: tabWidth } : { width: 'auto' };
```